test/index.test.js

```diff
@@ -32,32 +32,51 @@ describe('aws-budget-alert-lambda', function() {
 
             done()
         }
-        
     })
 
     it( `should send event message to WEBHOOK_URL when WEBHOOK_URL environment variable defined and no error: `, function( done ) {
         const rpStub = sinon.stub(request, 'post') // mock rp.post() calls
 
         // calls to rp.post() return a promise that will resolve to an object
         rpStub.returns(Bluebird.resolve('ok'))
-        
+
         process.env.WEBHOOK_URL = WEBHOOK_URL
-       
+
         myLambda.handler( event_message, { /* context */ }, (err, result) => {
-        
+
             try {
-                
                 expect( err ).to.not.exist
-                verifyRequestSentToWebhook(rpStub)
-                        
+                verifyRequestSentToWebhook(rpStub, '')
+
                 done()
             }
             catch( error ) {
                 done( error )
             }
         })
+    })
+
+    it( `should send event message to WEBHOOK_URL when WEBHOOK_URL and MESSAGE_PREFIX environment variables defined and no error: `, function( done ) {
+        const rpStub = sinon.stub(request, 'post') // mock rp.post() calls
 
+        // calls to rp.post() return a promise that will resolve to an object
+        rpStub.returns(Bluebird.resolve('ok'))
+
+        process.env.WEBHOOK_URL = WEBHOOK_URL
+        process.env.MESSAGE_PREFIX = MESSAGE_PREFIX
+
+        myLambda.handler( event_message, { /* context */ }, (err, result) => {
 
+            try {
+                expect( err ).to.not.exist
+                verifyRequestSentToWebhook(rpStub, MESSAGE_PREFIX)
+
+                done()
+            }
+            catch( error ) {
+                done( error )
+            }
+        })
```

Comment:
This way of testing is quite ugly but with the current implementation of the function I don't see a way around it.

Is it permissible for the function to return a promise that resolves on success and rejects on failure? That way you could use 'chai-as-promised' and use `result.should.be.fulfilled`.

Alternatively it might be possible to use Bluebird to 'promisify' your function that takes a callback.

Reply:
@drathore did most of the implementation.
@daniel-ac-martin would you be able to assist us to improve this?

Reply:
I'm wondering if we shouldn't do this as a separate ticket / PR considering there are still a few more PRs to review to fully deliver AWS budgets alerting?

Reply:
Perhaps we should leave it for now and raise an issue to improve it?

Reply:
#8